src-tauri/src/commands.rs

```diff
@@ -523,29 +523,13 @@ pub async fn get_monero_seed_words(
 #[tauri::command]
 pub async fn get_p2pool_stats(
     state: tauri::State<'_, UniverseAppState>,
-) -> Result<Option<Stats>, String> {
+) -> Result<Option<P2poolStats>, String> {
     let timer = Instant::now();
-    if state.is_getting_p2pool_stats.load(Ordering::SeqCst) {
-        let read = state.cached_p2pool_stats.read().await;
-        if let Some(stats) = &*read {
-            warn!(target: LOG_TARGET, "Already getting p2pool stats, returning cached value");
-            return Ok(stats.clone());
-        }
-        warn!(target: LOG_TARGET, "Already getting p2pool stats");
-        return Err("Already getting p2pool stats".to_string());
-    }
-    state.is_getting_p2pool_stats.store(true, Ordering::SeqCst);
-    let p2pool_stats = state.p2pool_manager.get_stats().await.unwrap_or_else(|e| {
-        warn!(target: LOG_TARGET, "Error getting p2pool stats: {}", e);
-        None
-    });
+    let p2pool_stats = state.p2pool_latest_status.borrow().clone();
```

Comment:
I've seen some issues with the latest status caching. When you stop mining the last mining status will be cached, and continue to offer whatever stats were last given. Even if no longer connected. (This was fixed for the GPU miner before). 

Just mentioning it here in case it's something that should be looked at in regards to p2p stats. Such as p2p is no longer connected but the cached stats have `connected_since` and makes everyone think it's still connected.

https://github.com/tari-project/universe/pull/1304/files#diff-31c51000768ebc0fedcd78a7c5282936ecee9a65108f5deff570b4d98af9a12cR123-R129

Reply:
That is a good point, although for p2pool it's not relevant since you can't turn off p2pool without restarting. Do you agree?

Reply:
That's fine with me for now. For some reason I though p2p got shut off if you weren't mining at all.